### ISSUE TITLE
Pagination Feature

### ISSUE THREAD

**Issue by zenull** [misterunknown/ifm#142]
Could you please add pagination feature so it can be possible browse directories with thousands of files?

Many thanks and much appreciated!

**Comment by misterunknown** [comment]
Yes. I'm working currently on an update which comes with some breaking changes. but is not ready yet.

**Comment by misterunknown** [comment]
If you don't want to wait for the next release, you can build the IFM from the branch "sane-coding-style" which I just pushed. There is an additional config option "paging", which is false by default, but can be set to true.

**Comment by SaswatPadhi** [comment]
I would love something like this as well. I have a few large directories and they take forever to load ...

BTW, thanks for creating and opensourcing this awesome project!

**Comment by misterunknown** [comment]
> I would love something like this as well. I have a few large directories and they take forever to load ...

Hm, the current solution works only client-side, which does  not reduce the loading time for the entire file table. Implementing a "lazy-loading" solution would take some more time.

**Comment by SaswatPadhi** [comment]
Thank you, @misterunknown.

Either full lazy loading, or lazy loading of metadata (size, permissions, etc.) would be great for large directories.

**Comment by misterunknown** [comment]
I think I'll go with full lazy loading. Lazy loading only for the metadata would require some huge effort on the client-side and I'm really a JS noob. I hope I can use the DataTables library to handle it client-side, and hence only have to implement it server-side.

**Comment by novashdima** [comment]
Bug: when pagination enabled - image preview work only on 1-st page.
